nexus/api.go

```diff
@@ -176,3 +199,65 @@ func addContextTimeoutToHTTPHeader(ctx context.Context, httpHeader http.Header)
 	httpHeader.Set(HeaderRequestTimeout, time.Until(deadline).String())
 	return httpHeader
 }
+
+type Link struct {
+	// URL encoded information about the link.
+	URL string `json:"url"`
+	// Type can describe an actual data type for decoding the URL.
+	Type string `json:"type"`
+}
+
+func encodeLink(link Link) string {
+	return fmt.Sprintf("<%s>; Type=%q", link.URL, link.Type)
+}
+
+func decodeLink(encodedLink string) (Link, error) {
+	consumeUntilAndDiscard := func(buf []byte, c byte) (bool, []byte, []byte) {
+		idx := bytes.IndexByte(buf, c)
+		if idx == -1 {
+			return false, buf, nil
+		}
+		return true, bytes.TrimSpace(buf[:idx]), bytes.TrimSpace(buf[idx+1:])
+	}
+
+	var link Link
+	buf := []byte(encodedLink)
```

Comment:
Why convert to `[]byte`?

Reply:
Re-wrote the function assuming that certain chars are forbidden. It simplified the implementation.